packages/client/src/tools/logging.ts

```diff
@@ -0,0 +1,66 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2024 TypeFox and others.
+ * Licensed under the MIT License. See LICENSE in the package root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { ConsoleLogger as VSCodeConsoleLogger, ILogger } from '@codingame/monaco-vscode-log-service-override';
+import { LogLevel } from 'vscode/services';
+
+export interface Logger extends ILogger {
+    createErrorAndLog(message: string, ...params: unknown[]): Error;
+}
+
+export class ConsoleLogger extends VSCodeConsoleLogger {
+
+    constructor(logLevel?: LogLevel, useColors?: boolean) {
+        super(logLevel ?? LogLevel.Off, useColors);
+    }
+
+    createErrorAndLog(message: string, ...params: unknown[]) {
+        if (this.getLevel() >= LogLevel.Error) {
+            this.error(message, ...params);
+        }
+        return new Error(message);
+    }
+
+}
+
+export function checkLogLevel(logLevel?: string | number | unknown) {
+    switch (logLevel) {
+        case 'none':
+        case 'Off':
+        case 'off':
+        case '0':
+        case 0:
+            return LogLevel.Off;
+        case 'Trace':
+        case 'trace':
+        case '1':
+        case 1:
+            return LogLevel.Trace;
+        case 'Debug':
+        case 'debug':
+        case '2':
+        case 2:
+            return LogLevel.Debug;
+        case 'Info':
+        case 'info':
+        case '3':
+        case 3:
+            return LogLevel.Info;
+        case 'Warning':
+        case 'warning':
+        case 'Warn':
+        case 'warn':
+        case '4':
+        case 4:
+            return LogLevel.Warning;
+        case 'Error':
+        case 'error':
+        case '5':
+        case 5:
+            return LogLevel.Error;
+        default:
+            return LogLevel.Off;
+    }
```

Comment:
why do we need to handle all these values ? (:

Reply:
I don't get it. It's just a helper function.

Reply:
Yes, I'm not sure why we need a function that takes anything as parameter and tries its best to understand what it's about, instead of having strong types?

Reply:
I don't have to `import { LogLevel } from 'vscode/services';` in all examples, but maybe you are right. It makes sense when you want to parse command-line values, but we don't need to do this here.

Reply:
Done ⬇️